### DISCUSSION TITLE
Pass already extracted `BufferVec` on the gpu from compute shader into fragment shader

### DISCUSSION THREAD

**Discussion by 68317fa2** [bevyengine/bevy#14797] [Q&A] [answered]
I am sorry for my missing understanding of bevy's rendering. So I have a compute shader, that gets a `texture_storage_2d` and outputs a `BufferVec` based on the input. Then, I want to pass this buffer into a fragment shader, that transforms the color of the faces of a mesh, that has a given material. I need to partially copy the `BufferVec` from the gpu back to the cpu, so I cannot directly pass the `texture_storage_2d` into the fragment shader. I already have a `RenderPipeline` for creating a `BindGroupLayout` and `BindGroup` for the compute shader and I wonder, how I can manually pass the `BufferVec` into the fragment shader after that. Note that I cannot copy the very large `BufferVec` onto the cpu and do the "normal" `Material` workflow shown in the [Array Texture Example](https://bevyengine.org/examples/shaders/array-texture/). I'd be very happy to get some links/other resources on the whole material workflow or a tutorial that covers my question. I also understand that this might be the wrong place to learn how to do things like that, so I'd appreciate tips/links to other forums/blogs/resources.

**Comment by 68317fa2** [comment, marked as answer]
Ok, found the answer! As always, it is just a problem of knowledge and familiarity with bevy and its rendering/shader system.

**Reply by thmxv** [reply]
Hi,

I am looking in how I can pass the Mesh data (attributes and indices) to a compute shader. Have the compute shader generate another mesh geometry and pass that to the fragment shader. Have you any link or documentation on how to do this?

**Reply by Touma-Kazusa2** [reply]
> Hi,
> 
> I am looking in how I can pass the Mesh data (attributes and indices) to a compute shader. Have the compute shader generate another mesh geometry and pass that to the fragment shader. Have you any link or documentation on how to do this?

sounds like meshletPlugin https://docs.rs/bevy/latest/bevy/pbr/experimental/meshlet/struct.MeshletPlugin.html
